### PR TITLE
[netif] remove multicast promiscuous mode configuration

### PR DESCRIPTION
**Pull request by konradderda**
Since 74c833b6236ac1dab3673fb8ebd705eca662d2b4 was introduced all IPv6 multicast packets are passed to the host. Previously used promiscuous mode configuration is now unused. This commit removes the dead code and corresponding elements as there is no use case for them at the moment (and they can confuse a user).

### CODE REVIEW

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -3365,35 +3365,6 @@ template <> otError Interpreter::Process<Cmd("ipmaddr")>(Arg aArgs[])
...
-     * @cli ipmaddr promiscuous
-     * @code
-     * ipmaddr promiscuous
-     * Disabled
-     * Done
-     * @endcode
-     * @par api_copy
-     * #otIp6IsMulticastPromiscuousEnabled
-     */
-    else if (aArgs[0] == "promiscuous")
```
Comment (abtink) [inline src/cli/cli.cpp:3378]: We should remove the use of the `ipmaddr promiscuous` command across the codebase. It is currently found in:
*   `src/cli/README.md`
*   `tests/scripts/expect/cli-ipmaddr.exp`
*   `tools/otci/otci.py` and `tools/otci/tests/test_otci.py`

I used `git grep "ipmaddr promiscuous"` to search for it:
```bash
abtink ~/github/openthread (main) $ git grep "ipmaddr promiscuous"
src/cli/README.md:### ipmaddr promiscuous
src/cli/README.md:> ipmaddr promiscuous
src/cli/README.md:### ipmaddr promiscuous enable
src/cli/README.md:> ipmaddr promiscuous enable
src/cli/README.md:### ipmaddr promiscuous disable
src/cli/README.md:> ipmaddr promiscuous disable
src/cli/cli.cpp:     * @cli ipmaddr promiscuous
src/cli/cli.cpp:     * ipmaddr promiscuous
src/cli/cli.cpp:         * @cli ipmaddr promiscuous (enable,disable)
src/cli/cli.cpp:         * ipmaddr promiscuous enable
src/cli/cli.cpp:         * ipmaddr promiscuous disable
src/cli/cli.cpp:         * @cparam ipmaddr promiscuous @ca{enable|disable}
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous enable\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous disable\n"
tests/scripts/expect/cli-ipmaddr.exp:send "ipmaddr promiscuous\n"
tools/otci/otci/otci.py:        return self.__parse_Enabled_or_Disabled(self.execute_command("ipmaddr promiscuous"))
tools/otci/otci/otci.py:        self.execute_command('ipmaddr promiscuous enable')
tools/otci/otci/otci.py:        self.execute_command('ipmaddr promiscuous disable')
tools/otci/tests/test_otci.py:        logging.info("ipmaddr promiscuous: %r", leader.get_ipmaddr_promiscuous())
```

**Review by abtink — COMMENTED** [review]
Thanks @konradderda for doing this. 👍 
Looks good overall.

**Comment by konradderda** [comment]
@abtink thanks for pointing this out! I have removed other references to the command

**Review by abtink — APPROVED** [review]
Thanks. LGTM. 👍

**Comment by jwhui** [comment]
@konradderda , a few checks are failing. Can you help take a look?

**Comment by konradderda** [comment]
@jwhui There was a one line to be removed in test scripts. Now it's fixed

**Pull request merged**